app/lib/katello/capsule_content.rb

```diff
@@ -151,6 +151,12 @@ def pulp_repo_facts(pulp_id)
       nil
     end
 
+    def ping_pulp
+      ::Katello::Ping.pulp_without_auth(self.pulp_url)
+    rescue Errno::EHOSTUNREACH
+      raise ::Katello::Errors::CapsuleCannotBeReached, _("%s is unreachable" % @capsule.name)
+    end
```

Comment:
This doesn't seem to work if the capsule is reachable but the services aren't up.  Such as if httpd and/or foreman-proxy are down.  You might wanna also catch RestClient::Exception